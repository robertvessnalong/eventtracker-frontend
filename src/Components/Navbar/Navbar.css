.Navbar {
  position: absolute;
  width: 220px;
}

.Mobile {
  width: 80px !important;
  min-width: 80px !important;
}
.Navbar .pro-sidebar {
  height: 100vh;
  width: 100%;
  min-width: 100%;
}
.Navbar .closemenu {
  color: #000;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 9999;
  line-height: 20px;
  border-radius: 50%;
  font-weight: bold;
  font-size: 22px;
  cursor: pointer;
  padding: 15px;
}

.Navbar .pro-sidebar.collapsed {
  width: 80px;
  min-width: 80px;
}
.Navbar .pro-sidebar-inner {
  background-color: #cfd5da !important;
  box-shadow: 0.5px 0.866px 2px 0px rgba(0, 0, 0, 0.15);
}
.pro-sidebar-layout {
  overflow-y: hidden;
}
.pro-sidebar-layout h1 {
  font-size: 20px;
  padding: 15px 35px 8px 20px;
  color: #000;
  font-weight: bold;
  text-transform: uppercase;
}

.pro-sidebar-header {
  display: flex;
  justify-content: center;
}

.pro-sidebar-header a {
  text-decoration: none !important;
}

.pro-sidebar-layout ul {
  padding: 0 5px;
}
.pro-sidebar-layout ul .pro-inner-item {
  color: #000;
  margin: 10px 0px;
  font-weight: bold;
}
.pro-sidebar-layout ul .pro-inner-item:hover {
  color: #000;
}

.Navbar
  .pro-sidebar-inner
  .pro-sidebar-layout
  ul
  .pro-inner-item
  .pro-icon-wrapper {
  background-color: #fbf4cd !important;
  border-radius: 3px !important;
}

.Navbar
  .pro-sidebar-inner
  .pro-sidebar-layout
  ul
  .pro-inner-item
  .pro-icon-wrapper
  .pro-item-content {
  color: #000;
}

.Navbar .logo {
  padding: 20px;
}

.Navbar .pro-sidebar .pro-menu a {
  text-decoration: none;
  color: #000;
}

@media only screen and (max-width: 720px) {
  html {
    overflow: hidden;
  }
}
