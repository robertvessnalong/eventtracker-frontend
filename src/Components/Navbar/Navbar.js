import { useState, useContext, useEffect } from 'react';
import {
  ProSidebar,
  Menu,
  MenuItem,
  SidebarHeader,
  SidebarFooter,
  SidebarContent,
} from 'react-pro-sidebar';
import 'react-pro-sidebar/dist/css/styles.css';
import {
  FaRegCalendar,
  FaUserPlus,
  FaUserAlt,
  FaBuilding,
  FaMicrophone,
  FaUserCircle,
  FaComment,
  FaArrowAltCircleRight,
  FaArrowAltCircleLeft,
} from 'react-icons/fa';
import './Navbar.css';
import { NavLink, Link } from 'react-router-dom';
import UserContext from '../../Context/UserContext';

const Navbar = () => {
  const [isActive, setIsActive] = useState(false);
  const { user, logOut } = useContext(UserContext);
  const [loggedIn, setLoggedIn] = useState(false);

  useEffect(() => {
    Object.keys(user).length === 0 ? setLoggedIn(false) : setLoggedIn(true);
  }, [user]);

  const menuHandler = () => {
    isActive ? setIsActive(false) : setIsActive(true);
  };

  return (
    <>
      <div className={isActive ? `Navbar Mobile` : `Navbar`}>
        <ProSidebar collapsed={isActive}>
          <SidebarHeader>
            {!isActive && (
              <Link to='/'>
                <h1>Event Finder</h1>
              </Link>
            )}
            <div className='closemenu' onClick={menuHandler}>
              {/* changing menu collapse icon on click */}
              {isActive ? <FaArrowAltCircleRight /> : <FaArrowAltCircleLeft />}
            </div>
          </SidebarHeader>
          <SidebarContent>
            <Menu iconShape='square'>
              {loggedIn && (
                <>
                  <MenuItem icon={<FaRegCalendar />}>
                    <NavLink to='/events'>Events</NavLink>
                  </MenuItem>
                  <MenuItem icon={<FaMicrophone />}>
                    <NavLink to='/performers'>Performers</NavLink>
                  </MenuItem>
                  <MenuItem icon={<FaBuilding />}>
                    <NavLink to='/venues'>Venues</NavLink>
                  </MenuItem>
                  <MenuItem icon={<FaComment />}>
                    <NavLink to='/social'>Social</NavLink>
                  </MenuItem>
                  <MenuItem icon={<FaUserCircle />}>
                    <NavLink to='/profile'>Profile</NavLink>
                  </MenuItem>
                </>
              )}
              {!loggedIn && (
                <>
                  <MenuItem icon={<FaUserPlus />}>
                    <NavLink to='/signup'>Sign Up</NavLink>
                  </MenuItem>
                  <MenuItem icon={<FaUserAlt />}>
                    <NavLink to='/login'>Log In</NavLink>
                  </MenuItem>
                </>
              )}
            </Menu>
          </SidebarContent>
          <SidebarFooter>
            {loggedIn && (
              <Menu iconShape='square'>
                <MenuItem>
                  <NavLink onClick={logOut} to='/'>
                    Log Out
                  </NavLink>
                </MenuItem>
              </Menu>
            )}
          </SidebarFooter>
        </ProSidebar>
      </div>
    </>
  );
};

export default Navbar;
