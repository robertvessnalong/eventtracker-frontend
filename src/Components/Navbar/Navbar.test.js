import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Navbar from './Navbar';
import TestUser from '../../Testing/TestUser';
import UserContext from '../../Context/UserContext';
import { BrowserRouter } from 'react-router-dom';

describe('Events Component', () => {
  // Smoke Test
  test('renders without crashing', async () => {
    render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <Navbar />
        </UserContext.Provider>
      </BrowserRouter>
    );

    const h1 = screen.getByText(/Event Finder/i);
    expect(h1).toBeInTheDocument();
  });

  // Snapshot
  test('matches snapshot', async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <Navbar />
        </UserContext.Provider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
