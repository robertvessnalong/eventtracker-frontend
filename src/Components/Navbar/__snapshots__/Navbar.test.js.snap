// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Navbar"
  >
    <div
      class="pro-sidebar"
    >
      <div
        class="pro-sidebar-inner"
      >
        <div
          class="pro-sidebar-layout"
        >
          <div
            class="pro-sidebar-header"
          >
            <a
              href="/"
            >
              <h1>
                Event Finder
              </h1>
            </a>
            <div
              class="closemenu"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm116-292H256v-70.9c0-10.7-13-16.1-20.5-8.5L121.2 247.5c-4.7 4.7-4.7 12.2 0 16.9l114.3 114.9c7.6 7.6 20.5 2.2 20.5-8.5V300h116c6.6 0 12-5.4 12-12v-64c0-6.6-5.4-12-12-12z"
                />
              </svg>
            </div>
          </div>
          <div
            class="pro-sidebar-content"
          >
            <nav
              class="pro-menu shaped square"
            >
              <ul>
                <li
                  class="pro-menu-item"
                >
                  <div
                    class="pro-inner-item"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="pro-icon-wrapper"
                    >
                      <span
                        class="pro-icon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 448 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M400 64h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V112c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V160h352v298c0 3.3-2.7 6-6 6z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      class="pro-item-content"
                    >
                      <a
                        href="/events"
                      >
                        Events
                      </a>
                    </span>
                  </div>
                </li>
                <li
                  class="pro-menu-item"
                >
                  <div
                    class="pro-inner-item"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="pro-icon-wrapper"
                    >
                      <span
                        class="pro-icon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 352 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      class="pro-item-content"
                    >
                      <a
                        href="/performers"
                      >
                        Performers
                      </a>
                    </span>
                  </div>
                </li>
                <li
                  class="pro-menu-item"
                >
                  <div
                    class="pro-inner-item"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="pro-icon-wrapper"
                    >
                      <span
                        class="pro-icon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 448 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M436 480h-20V24c0-13.255-10.745-24-24-24H56C42.745 0 32 10.745 32 24v456H12c-6.627 0-12 5.373-12 12v20h448v-20c0-6.627-5.373-12-12-12zM128 76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76zm0 96c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40zm52 148h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12zm76 160h-64v-84c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v84zm64-172c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      class="pro-item-content"
                    >
                      <a
                        href="/venues"
                      >
                        Venues
                      </a>
                    </span>
                  </div>
                </li>
                <li
                  class="pro-menu-item"
                >
                  <div
                    class="pro-inner-item"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="pro-icon-wrapper"
                    >
                      <span
                        class="pro-icon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 512 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      class="pro-item-content"
                    >
                      <a
                        href="/social"
                      >
                        Social
                      </a>
                    </span>
                  </div>
                </li>
                <li
                  class="pro-menu-item"
                >
                  <div
                    class="pro-inner-item"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="pro-icon-wrapper"
                    >
                      <span
                        class="pro-icon"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 496 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      class="pro-item-content"
                    >
                      <a
                        href="/profile"
                      >
                        Profile
                      </a>
                    </span>
                  </div>
                </li>
              </ul>
            </nav>
          </div>
          <div
            class="pro-sidebar-footer"
          >
            <nav
              class="pro-menu shaped square"
            >
              <ul>
                <li
                  class="pro-menu-item"
                >
                  <div
                    class="pro-inner-item"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="pro-item-content"
                    >
                      <a
                        aria-current="page"
                        class="active"
                        href="/"
                      >
                        Log Out
                      </a>
                    </span>
                  </div>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
      <div
        aria-label="overlay"
        class="overlay"
        role="button"
        tabindex="0"
      />
    </div>
  </div>
</DocumentFragment>
`;
